blob.go

```diff
@@ -460,12 +461,18 @@ type blobGCHandler struct {
 	candidateDiscardSize uint64
 }
 
-func (h *blobGCHandler) run() {
-	for discardInfo := range h.discardCh {
-		h.handleDiscardInfo(discardInfo)
-		err := h.doGCIfNeeded()
-		if err != nil {
-			log.Error(err)
+func (h *blobGCHandler) run(c *y.Closer) {
+	defer c.Done()
+	for {
+		select {
+		case discardInfo := <-h.discardCh:
+			h.handleDiscardInfo(discardInfo)
+			err := h.doGCIfNeeded()
+			if err != nil {
+				log.Error(err)
+			}
+		case <-c.HasBeenClosed():
+			return
```

Comment:
Is it safe to ignore the `discardInfo` pending in `h.discardCh`? @coocood

Reply:
It is safe if we SignalAndWait it after compactors.